app/javascript/src/components/Navigation/components/GlobalActivitiesModal.jsx

```diff
@@ -79,45 +80,52 @@ class GlobalActivitiesModal extends Component<Props, State> {
   }
 
   onCloseModalActions(): void {
-    this.setState({ activities: [], more: false });
+    this.setState({ activities: [], more: false, currentPage: 1 });
     this.props.onCloseModal();
   }
 
   loadData(): void {
     getActivities().then(response => {
       this.setState({
         activities: response.activities,
-        more: response.more
+        more: response.more,
+        currentPage: response.currentPage
       });
     });
   }
 
   mapActivities(): Array<*> {
-    return this.state.activities.map((activity, i, arr) => {
-      // @todo replace key={i} with key={activity.id} !!!!!!!!!!!!!!
-      // when the backend bug will be fixed
-      const newDate = new Date(activity.created_at);
-      // returns a label with "today" if the date of the activity is today
-      if (i === 0) {
-        return GlobalActivitiesModal.renderActivityDateElement(
-          i,
-          activity,
-          newDate
-        );
+    return this.state.activities.map(
+      (activity: Activity, i: number, arr: Array<*>) => {
+        // when the backend bug will be fixed
```

Comment:
Which backend bug?

Reply:
a :banana: bug we have some custom heavy :banana: pagination that returned the same records multiple time I hope that kaminari gem will paginate properly